expander.go

```diff
@@ -369,6 +382,59 @@ func normalizePaths(refPath, base string) string {
 	return baseURL.String()
 }
 
+// denormalizePaths returns to simplest notation on file $ref,
+// i.e. strips the absolute path and sets a path relative to the base path.
+//
+// This is currently used when we rewrite ref after a circular ref has been detected
+func denormalizeFileRef(ref *Ref, relativeBase, originalRelativeBase string) *Ref {
```

Comment:
This is more complicated than expected at first. I tested it against JSONSchema schema, swagger schema to make sure resolved $ref are okay.